### PR TITLE
[ci][distributed] merge distributed test commands

### PR DESCRIPTION
**Pull request by youkaichao**
step 2 of https://github.com/vllm-project/vllm/issues/7053

### CODE REVIEW

File: .buildkite/test-pipeline.yaml [hunk 1]
```diff
@@ -107,11 +102,6 @@ steps:
   fast_check: true
   commands:
   - pytest -v -s distributed/test_pynccl.py
-  # We want to test that models which use 2 GPUs work with 4 GPUs, which is why we duplicate them here.
-  # See https://github.com/vllm-project/vllm/pull/5473#issuecomment-2166601837 for context.
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray VLLM_USE_RAY_SPMD_WORKER=1 VLLM_USE_RAY_COMPILED_DAG=1 pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=mp pytest -v -s distributed/test_basic_distributed_correctness.py
```
Comment (youkaichao) [inline .buildkite/test-pipeline.yaml:114]: these are not necessary after https://github.com/vllm-project/vllm/pull/7069

File: .buildkite/test-pipeline.yaml [hunk 2]
```diff
@@ -82,18 +82,10 @@ steps:
   num_gpus: 2
   commands:
   - VLLM_TEST_SAME_HOST=1 torchrun --nproc-per-node=4 distributed/test_same_node.py
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=meta-llama/Llama-2-7b-hf DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray VLLM_USE_RAY_SPMD_WORKER=1 VLLM_USE_RAY_COMPILED_DAG=1 pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=meta-llama/Llama-2-7b-hf DISTRIBUTED_EXECUTOR_BACKEND=ray VLLM_USE_RAY_SPMD_WORKER=1 VLLM_USE_RAY_COMPILED_DAG=1 pytest -v -s distributed/test_basic_distributed_correctness.py
-  - TEST_DIST_MODEL=facebook/opt-125m DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_chunked_prefill_distributed.py
-  - TEST_DIST_MODEL=meta-llama/Llama-2-7b-hf DISTRIBUTED_EXECUTOR_BACKEND=ray pytest -v -s distributed/test_chunked_prefill_distributed.py
+  - TARGET_TEST_SUITE=L4 pytest -v -s distributed/test_basic_distributed_correctness.py
```
Comment (DarkLight1337) [inline .buildkite/test-pipeline.yaml:85]: What does `TARGET_TEST_SUITE` do? I couldn't find it in the code.

File: tests/distributed/test_basic_distributed_correctness.py [hunk 3]
```diff
@@ -19,27 +14,48 @@
...
+TARGET_TEST_SUITE = os.environ.get("TARGET_TEST_SUITE", "L4")
 
 
 @pytest.mark.skipif(cuda_device_count_stateless() < 2,
                     reason="Need at least 2 GPUs to run the test.")
-@pytest.mark.parametrize("model", MODELS)
-@pytest.mark.parametrize("dtype", ["half"])
-@pytest.mark.parametrize("max_tokens", [5])
+@pytest.mark.parametrize(
+    "model, distributed_executor_backend, attention_backend, test_suit", [
```
Comment (DarkLight1337) [inline tests/distributed/test_basic_distributed_correctness.py:25]:
```suggestion
    "model, distributed_executor_backend, attention_backend, test_suite", [
```

File: .buildkite/test-pipeline.yaml (hunk 2 above)
Comment (youkaichao) [reply .buildkite/test-pipeline.yaml:85]: it is used here https://github.com/vllm-project/vllm/pull/7097/files#diff-f0ca36a332eaf5269b0c3a31e3f08e7e2105edcdd3605f7a383a08e4ec270b5bR19

File: tests/distributed/test_basic_distributed_correctness.py (hunk 3 above)
Comment (youkaichao) [reply tests/distributed/test_basic_distributed_correctness.py]: nice catch!

**Pull request merged**